packages/core/lib/util/get-response.ts

```diff
@@ -0,0 +1,139 @@
+import { ExecutionResult } from "graphql";
+import { ExecutionPatchResult } from "../types";
+import { calculateByteLength } from "./calculate-byte-length";
+import { ReadableStream, Response } from "cross-undici-fetch";
+
+export type TransformResultFn = (result: ExecutionResult | ExecutionPatchResult) => any;
+export const DEFAULT_TRANSFORM_RESULT_FN: TransformResultFn = (result: ExecutionResult) => result;
+
+export function getRegularResponse(executionResult: ExecutionResult, transformResult = DEFAULT_TRANSFORM_RESULT_FN): Response {
+  const transformedResult = transformResult(executionResult);
+  const responseBody = JSON.stringify(transformedResult);
+  const contentLength = calculateByteLength(responseBody);
+  const headersInit: HeadersInit = {
+    "Content-Type": 'application/json',
+    "Content-Length": contentLength.toString()
+  };
+  const responseInit: ResponseInit = {
+    headers: headersInit,
+    status: 200,
+  };
+  return new Response(responseBody, responseInit);
+}
+
+export function getMultipartResponse(
+  asyncExecutionResult: AsyncIterable<ExecutionPatchResult<any>>,
+  transformResult = DEFAULT_TRANSFORM_RESULT_FN
+): Response {
+  const headersInit: HeadersInit = {
+    Connection: "keep-alive",
+    "Content-Type": 'multipart/mixed; boundary="-"',
+    "Transfer-Encoding": "chunked",
+  };
+  const responseInit: ResponseInit = {
+    headers: headersInit,
+    status: 200,
+  };
+  const readableStream = new ReadableStream({
+    async start(controller) {
+      try {
+        controller.enqueue(`---`);
+        const iterator = asyncExecutionResult[Symbol.asyncIterator]();
+        while (true) {
+          const { done, value } = await iterator.next();
+          if (done) {
+            controller.enqueue("\r\n-----\r\n");
+            controller.close();
+            break;
+          }
+          const transformedResult = transformResult(value);
+          const chunk = JSON.stringify(transformedResult);
+          const contentLength = calculateByteLength(chunk);
+          const data = [
+            "",
+            "Content-Type: application/json; charset=utf-8",
+            "Content-Length: " + contentLength.toString(),
```

Comment:
```suggestion
          const data = [
            "",
            "Content-Type: application/json; charset=utf-8",
```

content-lengths in multipart segments isnt required, and isnt in the http graphql spec either.

Reply:
I think it is required in some env. 
https://github.com/contra/graphql-helix/pull/145#issue-1055975637

Reply:
Not in multipart responses, those are effectively "body" and is no mans land. The spec clearly stipulates its not required. Had the exact same learnings during my build of meros.

check https://datatracker.ietf.org/doc/html/rfc1341

Reply:
It is not spec but the HTTP Client itself. Not required doesn't mean we shouldn't provide it. I think it is good to have this.

Reply:
Its not good, you have no grounds for that. Multipart isnt part of the http header anymore. Its 100% superficial body.

Reply:
This is not something this PR covers. This part was already there before this PR.